flume-ng-sources/flume-scribe-source/src/main/java/org/apache/flume/source/scribe/ScribeSource.java

```diff
@@ -48,71 +48,158 @@
+  private static final int DEFAULT_THREADED_SELECTOR_NUM_THREADS = 4;
+  private static final long DEFAULT_MAX_READ_BUFFER_BYTES = 16384000;
+  private static final int DEFAULT_MAX_THRIFT_FRAME_SIZE_BYTES = 16384000;
 
   private TServer server;
   private int port;
   private int workers;
-  private int maxReadBufferBytes;
+  private int selectorThreads;
+  private ThriftServerType thriftServerType;
+  private long maxReadBufferBytes;
+  private int maxThriftFrameSizeBytes;
 
   private SourceCounter sourceCounter;
 
   @Override
   public void configure(Context context) {
     port = context.getInteger("port", DEFAULT_PORT);
-    maxReadBufferBytes = context.getInteger("maxReadBufferBytes", DEFAULT_MAX_READ_BUFFER_BYTES);
-    if(maxReadBufferBytes <= 0){
+    maxReadBufferBytes = context.getLong("maxReadBufferBytes", DEFAULT_MAX_READ_BUFFER_BYTES);
+    if (maxReadBufferBytes <= 0) {
       maxReadBufferBytes = DEFAULT_MAX_READ_BUFFER_BYTES;
     }
 
+    maxThriftFrameSizeBytes = context.getInteger("maxThriftFrameSizeBytes",
+        DEFAULT_MAX_THRIFT_FRAME_SIZE_BYTES);
+    if (maxThriftFrameSizeBytes <= 0) {
+      maxThriftFrameSizeBytes = DEFAULT_MAX_THRIFT_FRAME_SIZE_BYTES;
+    }
+
     workers = context.getInteger("workerThreads", DEFAULT_WORKERS);
     if (workers <= 0) {
       workers = DEFAULT_WORKERS;
     }
 
+    try {
+      thriftServerType = ThriftServerType.getServerType(
+          context.getString("thriftServer", DEFAULT_THRIFT_SERVER_TYPE.getValue()));
+    } catch (IllegalArgumentException ex) {
+      LOG.error("Wrong thrift server type config", ex);
+    }
+    selectorThreads = context.getInteger("selectorThreads", DEFAULT_THREADED_SELECTOR_NUM_THREADS);
+    if (selectorThreads <= 0) {
+      selectorThreads = DEFAULT_THREADED_SELECTOR_NUM_THREADS;
+    }
+
     if (sourceCounter == null) {
       sourceCounter = new SourceCounter(getName());
     }
   }
 
   private class Startup extends Thread {
-
     public void run() {
       try {
+        if (thriftServerType == null) {
+          LOG.error("Unknow thrift server type configured, " +
+              "it should be one of {THshaServer, TThreadedSelectorServer, TThreadPoolServer}");
+          return;
+        }
         Scribe.Processor processor = new Scribe.Processor(new Receiver());
         TNonblockingServerTransport transport = new TNonblockingServerSocket(port);
-        THsHaServer.Args args = new THsHaServer.Args(transport);
-
-        args.minWorkerThreads(workers);
-        args.maxWorkerThreads(workers);
-        args.processor(processor);
-        args.transportFactory(new TFramedTransport.Factory(maxReadBufferBytes));
-        args.protocolFactory(new TBinaryProtocol.Factory(false, false));
-        args.maxReadBufferBytes = maxReadBufferBytes;
-
-        server = new THsHaServer(args);
-
+        LOG.info("Initializing thrift server:" + thriftServerType.getValue());
+        switch (thriftServerType) {
```

Comment:
maybe this could be in a factory to make it readable and more organized.
 
We could expose the ThriftServer configuration parameters in the source params as in the HTTPSource with the FlumeBeanConfigurator
https://github.com/apache/flume/blob/trunk/flume-ng-core/src/main/java/org/apache/flume/source/http/HTTPSource.java#L198

Reply:
Sure, will update it later